httputil/retry.go

```diff
@@ -127,6 +135,57 @@ func ShouldRetryError(attempt *retry.Attempt, err error) bool {
 	return false
 }
 
+func isNetworkDown(err error) bool {
+	urlErr, ok := err.(*url.Error)
+	if !ok {
+		return false
+	}
+	opErr, ok := urlErr.Err.(*net.OpError)
```

Comment:
All the error peeling is meh. Wish we had other ways of doing this without pulling in extra dependencies.

Reply:
Amen, brother! Can't agree more!

Reply:
/me nods